go.mod

```diff
@@ -2,6 +2,9 @@ module github.com/openstack-k8s-operators/glance-operator
 
 go 1.19
 
+replace github.com/openstack-k8s-operators/mariadb-operator/api => /home/classic/dev/openshift/mariadb-operator/api
```

Comment:
I've only run this locally so far.   hmmm I guess I can link to the branch but i wanted to get feedback on the code before we try to test this etc.

Reply:
I think it make sense to link it to your mariadb PR and see the test results.

Reply:
how do you do that?  I saw you did it the other day but I cant find that commit now

Reply:
got it, hard to google for

Reply:
OK this is not passing any tests if you could show me how to do this that would be helpful

Reply:
I suspect you should go get https://github.com/zzzeek/mariadb-operator/api@integrate_oo_accounts  and put here `https://github.com/zzzeek/mariadb-operator/api <the resulting hash>`

I didn't review this patch entirely but it would be a good start to have something that passes checks, so we can use logs to understand the bits involved here.
The problem I see is that the job that is supposed to test this particular concatenation of patches can only rebuild glance and the openstack-operator, so we should inject the right mariadb image that is able to handle the new code (not sure we're able to do that in CI right now).
Maybe we can do that by patching the openstack-operator (where we inject an arbitrary mariadb-operator image) and add the `Depends-On` to this patch.
That probably would allow the test the whole chain.
The steps should be:
1. rebuild the operator based on https://github.com/openstack-k8s-operators/mariadb-operator/pull/184 and push it to a private quay.io (make sure to set the image as public) (together w/ the associated bundle and index)
2. hack the openstack-operator script (https://github.com/openstack-k8s-operators/openstack-operator/blob/main/hack/pin-bundle-images.sh) to build it using your quay.io bundle pushed at step1
3. add to this patch a `Depends-on` to the openstack-operator patch created at step2.

I did something similar in the past to just see the CI green, and those are pretty much the steps (we might need a variation of them though).

Reply:
the go get approach gives me this error:

```
$ go get github.com/zzzeek/mariadb-operator/api@integrate_oo_accounts
go: downloading github.com/zzzeek/mariadb-operator/api v0.3.2
go: downloading github.com/zzzeek/mariadb-operator v0.3.2
go: github.com/zzzeek/mariadb-operator/api@v0.3.2: parsing go.mod:
        module declares its path as: github.com/openstack-k8s-operators/mariadb-operator/api
                but was required as: github.com/zzzeek/mariadb-operator/api

```

The changes on the mariadb-operator side are just new API additions within the `mariadbdatabase_funcs.go` file, and otherwise dont change how that operator works at all, so we may consider that I'd test this some more locally enough that we can merge just the mariadb side of it.    I can go the quay approach here but this looks like a steep hill to climb